src/main/scala/DateFormat.scala

```diff
@@ -1,127 +1,69 @@
-//        case Some(_) =>
-//          val maxFractionalSeconds = pattern.split(".").last.length()
-//          regExp = (regExp.toString().replaceFirst("/ss\\.S+$/", raw"""(?<second>$fields("ss")(\\.[0-9]{1,$fields})?)""")).r
-//        case None => regExp = (regExp.toString().replaceFirst("ss", raw"""(?<second>$fields("ss")""")).r
-//      }
-//
-//      Option("/yyyy/".r.findAllMatchIn(regExp.toString()).toArray) match {
-//        case Some(_) =>
-//          regExp = (regExp.toString().replaceFirst("yyyy", fields("yyyy").toString())).r
-//          regExp = (regExp.toString().replaceFirst("MM", fields("MM").toString())).r
-//          regExp = (regExp.toString().replaceFirst("M", fields("M").toString())).r
-//          regExp = (regExp.toString().replaceFirst("dd", fields("dd").toString())).r
-//          regExp = (regExp.toString().replaceFirst("/d(?=[-T \\/\\.])/", fields("d").toString())).r
-//      }
-//
-//      regExp = (regExp.toString().replaceFirst("XXX", fields("XXX").toString())).r
-//      regExp = (regExp.toString().replaceFirst("XX", fields("XX").toString())).r
-//      regExp = (regExp.toString().replaceFirst("X", fields("XXX").toString())).r
-//      regExp = (regExp.toString().replaceFirst("xxx", fields("xxx").toString())).r
-//      regExp = (regExp.toString().replaceFirst("xx", fields("xx").toString())).r
-//      regExp = (regExp.toString().replaceFirst("/x(?!:)/", fields("d").toString())).r
-//
-//      regExp = raw"""^$regExp$$""".r
-//    }
-//  }
-//
-//  def parse(value: String): HashMap[String, Any]= {
-//    throw new NotImplementedError()
-//  }
-//}
\ No newline at end of file
+package CSVValidation
+import Errors.DateFormatError
+
+case class DateFormat(pattern: Option[String], dataType: Option[String]) {
+  private var `type`: String = _
+  private var format: String = _
+  private val fields = Array[String](
+    "yyyy",
+    "M",
+    "MM",
+    "dd",
+    "d",
+    "HH",
+    "mm",
+    "ss",
+    "X",
+    "XX",
+    "XXX",
+    "x",
+    "xx" ,
+    "xxx",
+  )
+  private val xmlSchemaBaseUrl = "http://www.w3.org/2001/XMLSchema#"
+
+  def getType() = `type`
+  def getFormat() = format
+
+  (pattern, dataType) match {
+    case (None, Some(dt)) => {
+      `type` = dt
+      format = "YYYY-MM-DDThh:mm:ss.sTZD"
+    }
+    case (Some(p), None) => {
+      format = p
+      ensureDateTimeFormatContainsRecognizedSymbols(p)
+      val hoursRegExp = "HH".r
+      val yearsRegExp = "yyyy".r
+      val includesHours = hoursRegExp.pattern.matcher(p).find()
+      val includesYears = yearsRegExp.pattern.matcher(p).find
+      `type` = if (!includesHours && includesYears) {
+        s"${xmlSchemaBaseUrl}date"
+      } else if (includesHours && !(includesYears)) {
+        s"${xmlSchemaBaseUrl}time"
+      } else if (includesHours && includesYears) {
+        s"${xmlSchemaBaseUrl}dateTime"
+      } else throw new DateFormatError("Dont know what exactly to perform in this case")
```

Comment:
I'd alter this message so it said something like "Unexpected datetime format '${p}' does not appear to contain date or time components."

Reply:
Done